### PR TITLE
More docs about keys.

### PR DESCRIPTION
**Pull request by Hixie**
I added a bunch of sample code to the core Widget subclasses so that
people would see that you are supposed to always include the `key`
argument.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/8818

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -483,6 +498,49 @@ abstract class Widget {
...
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return new Container(color: color, child: child);
+///   }
+/// }
+/// ```
+///
+/// By convention, widget constructors only use named arguments. Named arguments
+/// can be marked as required using [@required]. Also by convention, the first
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/framework.dart]: Does `[@required]` linkify correctly?

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/widgets/framework.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: No idea, I plan to check and fix it if not.

**Pull request merged**